Cql/PackagerCLI/DependencyInjection.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Runtime.InteropServices.ComTypes;
+using Hl7.Cql.Packaging;
+using Hl7.Cql.Packaging.ResourceWriters;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace Hl7.Cql.Packager;
+
+internal static class DependencyInjection
+{
+    public static void AddPackagerServices(this IServiceCollection services)
+    {
+        services.TryAddPackagerOptions();
+        services.TryAddTransient<PackagerService>();
+    }
+
+    private static void TryAddPackagerOptions(this IServiceCollection services)
+    {
+        if (services.Any(s => s.ServiceType == typeof(IValidateOptions<PackagerOptions>)))
+            return;
+
+        services.AddOptions<PackagerOptions>()
+            .BindConfiguration(PackagerOptions.ConfigSection)
+            .Configure<IConfiguration>((opt, config) => { PackagerOptions.BindDirectoryInfos(config, opt); })
+            .ValidateOnStart();
+        services.AddSingleton<IValidateOptions<PackagerOptions>, PackagerOptions.Validator>();
+    }
+
+    public static void AddResourcePackager(this IServiceCollection services)
+    {
+        services.TryAddPackagerOptions();
+        services.TryAddSingleton<ResourcePackager>();
+
+        var unvalidatedPackagerOptions = TryGetPackagerOptions(services);
```

Comment:
This feels like higher math to me!

Reply:
Are we setting up a nested container, to get to the options before they are really validated by the framework itself?

Reply:
(Ewout is now wondering whether we are doing such exotic things that this is not supported out of the box - maybe because of the DirectoryInfo settings?)

Reply:
I rewrote this so we dont have to create a serviceprovider during the process of describing the services.